src/pytorch_lightning/core/datamodule.py

```diff
@@ -182,8 +189,8 @@ def load_from_checkpoint(
         cls,
         checkpoint_path: Union[_PATH, IO],
         hparams_file: Optional[_PATH] = None,
-        **kwargs,
-    ):
+        **kwargs: Any,
+    ) -> Any:
```

Comment:
The return type is not correct

Reply:
I traced this to `pytorch_lightning.core.saving._load_from_checkpoint`; which calls `pytorch_lightning.core.saving._load_state` ... load_state also appears to have an incorrect return type of Any. Based on [obj be set to cls](https://github.com/Lightning-AI/lightning/blob/cd92e3541eab86a837b07515814c93e0ed1cac9f/src/pytorch_lightning/core/saving.py#L250) in core.saving._load_state, and cls being either `"pl.LightningModule", "pl.LightningDataModule"`, I think that `core.datamodule.load_from_checkpoint` `core.saving._load_from_checkpoint` and `core.saving._load_state` should all have returns of `"pl.LightningModule", "pl.LightningDataModule"` ...

Is that correct?

Reply:
That seems correct. Would you like to take `core.saving` to work on that first?

Reply:
sounds good to me; I'm going to create my working branch for `core.saving` from this branch, and once again label this branch as blocked by saving, if that is okay with you.